src/test/scala/com/databricks/labs/overwatch/pipeline/PipelineFunctionsTest.scala

```diff
@@ -185,4 +191,199 @@ class PipelineFunctionsTest extends AnyFunSpec with DataFrameComparer with Spark
+      assertResult(null) (df
+        .withColumn("epoch_milliseconds", PipelineFunctions.tsToEpochMilli("ts"))
+        .select("epoch_milliseconds")
+        .collect().head.get(0)
+      )
+
+    }
+  }
+
+  describe("Tests for getDeltaHistory") {
+    it("should get the history of delta table") {
+      Given("a delta table")
+      val df = spark.range(10).withColumn("c1", lit("c1"))
+      df.write.format("delta").mode("overwrite").save("/tmp/overwatch/tests/table1")
+
+      val dummyConfig = new Config()
+      dummyConfig.setDatabaseNameAndLoc("dummy", "dummy", "/tmp/overwatch/tests")
+
+      val testTable: PipelineTable = PipelineTable(name = "table1", _keys = Array("id"), config = dummyConfig)
+
+      When("function is called on the given delta table")
+      val functionOutputDf: DataFrame = PipelineFunctions.getDeltaHistory(spark, testTable)
+
+      Then("returns a dataframe with detal table history")
+      val expectedDf: DataFrame = DeltaTable.forPath(testTable.tableLocation).history(9999)
+        .select("version", "timestamp", "operation", "clusterId", "operationMetrics", "userMetadata")
+
+      assertApproximateDataFrameEquality(functionOutputDf, expectedDf, 0.001)
+
+      // delete the delta table
+      val dir = Directory("/tmp/overwatch/tests/table1")
+      dir.deleteRecursively()
+    }
+    it("should throw an exception when path is not a delta table") {
+      Given("a delta table")
+      val df = spark.range(10).withColumn("c1", lit("c1"))
+      df.write.format("delta").mode("overwrite").save("/tmp/overwatch/tests/table1")
+
+      val dummyConfig = new Config()
+      dummyConfig.setDatabaseNameAndLoc("dummy", "dummy", "/tmp/overwatch/test")
+
+      val testTable: PipelineTable = PipelineTable(name = "table1", _keys = Array("id"), config = dummyConfig)
+
+      When("function is called on incorrect delta table path")
+
+      Then("throws an exception")
+      assertThrows[org.apache.spark.sql.AnalysisException] (PipelineFunctions.getDeltaHistory(spark, testTable))
+    }
+  }
+
+  describe("Tests for getLastOptimized") {
+    it("should return 0 as long type when the table is never optimized") {
+      Given("a delta table")
+      val df = spark.range(10).withColumn("c1", lit("c1"))
+      df.write.format("delta").mode("overwrite").save("/tmp/overwatch/tests/table1")
+
+      val dummyConfig = new Config()
+      dummyConfig.setDatabaseNameAndLoc("dummy", "dummy", "/tmp/overwatch/tests")
+
+      val testTable: PipelineTable = PipelineTable(name = "table1", _keys = Array("id"), config = dummyConfig)
+
+      When("function is called on the delta table that is never optimized")
+
+      Then("return 0")
+      assertResult(0L) (PipelineFunctions.getLastOptimized(spark, testTable))
+
+      // delete the delta table
+      val dir = Directory("/tmp/overwatch/tests/table1")
+      dir.deleteRecursively()
+    }
+    it("should throw an exception when path is not a delta table") {
```

Comment:
This is a duplicate test underneath a different function. Should we create some generic test suite for common, reusable tests like this?